PatternPal/PatternPal.Extension/ViewModels/StepByStepListViewModel.cs

```diff
@@ -20,26 +26,152 @@ public class StepByStepListViewModel : ViewModel
 
         public ICommand NavigateHomeCommand { get; }
         public ICommand NavigateStepByStepInstructionsCommand { get; }
-        public IList< InstructionSet > InstructionSetList { get; set; }
-        public InstructionSet SelectedInstructionSet { get; set;  }
+        public ICommand NavigateContinueStepByStepInstructionsCommand { get; }
+        public IList< Recognizer > InstructionSetList { get; set; }
+        public Recognizer SelectedInstructionSet { get; set;  }
 
         public StepByStepListViewModel(
             NavigationStore navigationStore)
         {
             NavigateHomeCommand = new NavigateCommand< HomeViewModel >(
                 navigationStore,
                 () => new HomeViewModel(navigationStore));
+            
+            // Next button 
             NavigateStepByStepInstructionsCommand =
                 new NavigateCommand< StepByStepInstructionsViewModel >(
                     navigationStore,
-                    () => new StepByStepInstructionsViewModel(
-                        navigationStore,
-                        SelectedInstructionSet));
+                    () =>
+                    {
+                        return new StepByStepInstructionsViewModel(
+                            navigationStore,
+                            SelectedInstructionSet,
+                            StepByStepModes.New,
+                            CreateNewWorkFileInSolution());
+                    });
+            
+            // Continue button
+            NavigateContinueStepByStepInstructionsCommand =
+                new NavigateCommand<StepByStepInstructionsViewModel>(
+                    navigationStore,
+                    () =>
+                    {
+                        return new StepByStepInstructionsViewModel(
+                            navigationStore,
+                            SelectedInstructionSet,
+                            StepByStepModes.Continue,
+                            ContinueButtonBehavior());
+                    });
+
+            GetInstructionSetsResponse inStructionSetsResponse = 
+                GrpcHelper.StepByStepClient.GetInstructionSets(new GetInstructionSetsRequest());
+
+            InstructionSetList = inStructionSetsResponse.Recognizers;
+            SelectedInstructionSet = InstructionSetList.FirstOrDefault();
+        }
+
+        private List<string> ContinueButtonBehavior()
+        {
+            List<string> result = new List<string>();
+
+            OpenFileDialog ofd = new OpenFileDialog() { Multiselect = true, Filter = "cs files (*.cs)|*.cs", };
+
+            while (result.Count == 0)
+            {
+                bool? res = ofd.ShowDialog();
+                if (res.HasValue && res.Value)
+                {
+                    result = ofd.FileNames.ToList();
+                }
+                else
+                {
+                    MessageBox.Show("No files were provided!");
+                    ofd.ShowDialog();
```

Comment:
```suggestion
```
This is actually incorrect. You don't store the result of this call, which means that you trigger the `ShowDialog` call on line 81 after the user has closed this dialog, possibly 
resulting in us forgetting the files the user has selected.

Reply:
I have removed this showDialog.